### PR TITLE
Tenor GIF Integration for Atlas-Android-Messenger

### PR DESCRIPTION
**Pull request by ctrl-alt-del**
Tenor GIF Integration for Atlas-Android-Messenger

This PR also includes the fix for the 404 authentication issue on the Layer Identity Provider created using your latest Heroku script.

### CODE REVIEW

File: app/src/main/java/com/layer/messenger/util/LayerAuthenticationProvider.java [hunk 1]
```diff
@@ -160,10 +164,8 @@ private void respondToChallenge(LayerClient layerClient, String nonce) {
 
             // Credentials
             JSONObject rootObject = new JSONObject();
-            JSONObject userObject = new JSONObject();
-            rootObject.put("user", userObject);
-            userObject.put("email", credentials.getEmail());
-            userObject.put("password", credentials.getPassword());
+            rootObject.put("email", credentials.getEmail());
```
Comment (thecombatwombat) [inline app/src/main/java/com/layer/messenger/util/LayerAuthenticationProvider.java]: @ctrl-alt-del Not sure why you've made these changes here. Please let us know if there is a specific problem you're encountering, or revert this change.

File: app/src/main/java/com/layer/messenger/ConversationsListActivity.java [hunk 2]
```diff
@@ -38,6 +39,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // Atlas methods
         mConversationsList.init(getLayerClient(), getPicasso())
+                .addCellFactories(new ThreePartGifCellFactory(getLayerClient(), getGifLoaderClient()))
```
Comment (thecombatwombat) [inline app/src/main/java/com/layer/messenger/ConversationsListActivity.java]: Please group this in with the other arguments in the `.addCellFactories()` call below

File: app/src/main/java/com/layer/messenger/util/LayerAuthenticationProvider.java (hunk 1 above)
Comment (ctrl-alt-del) [reply app/src/main/java/com/layer/messenger/util/LayerAuthenticationProvider.java]: hey @thecombatwombat ;  Per my description on this PR, I was encountering a 404 authentication issue on the Layer Identity Provider created using your latest Heroku script.

It turned out the 404 is caused by two things (1) the incorrect structure of the POST parameters and (2) the missing `authenticate` route on the `url`.

This specific change that you are pointing at is to fix (1).

Your current code is constructing the parameter structure like this:
```
{
    "user": {
        "email": "...",
        "password": "...",
    },
    "nonce": "..."
}
```

However, your Identity Provider is expecting the parameters structure to be like this:
```
{
    "email": "...",
    "password": "...",
    "nonce": "..."
}
```

File: app/src/main/java/com/layer/messenger/ConversationsListActivity.java (hunk 2 above)
Comment (ctrl-alt-del) [reply app/src/main/java/com/layer/messenger/ConversationsListActivity.java]: Sure

File: app/src/main/java/com/layer/messenger/util/LayerAuthenticationProvider.java (hunk 1 above)
Comment (thecombatwombat) [reply app/src/main/java/com/layer/messenger/util/LayerAuthenticationProvider.java]: @ctrl-alt-del Yeah, I see your point. However, ☝️ is specific to your test environment, and each customer is going to have to implement that for themselves in a manner specific to their authentication methods anyway. 

For now, I'll ask you to revert those changes and keep them local to your test projects. 

We'll make a ticket to fix this and make it work better for everyone.

Comment (ctrl-alt-del) [reply app/src/main/java/com/layer/messenger/util/LayerAuthenticationProvider.java]: @thecombatwombat alright, changes reverted for `LayerAuthenticationProvider`;

**Comment by thecombatwombat** [comment]
@ctrl-alt-del Hi... I was just investigating the CircleCI failure. Looks like it is not set up to work with forks that well. 

`Suppressing export of environment variables DUMMY_GOOGLE_SERVICES_JSON_CONFIG fork PR builds`

Basically, we use a dummy `google-services.json` to allow builds to happen w/o potentially leaking real app-ids/keys. CircleCI does not want to do that with forks as it turns out ¯\_(ツ)_/¯ 

That's something _we_ should fix. Sorry about the confusion.

**Comment by ctrl-alt-del** [comment]
@thecombatwombat , ok, thanks for the clarification.  I updated the `Atlas-Android` commit per your request, and let me know if you guys need anything else from me.

**Comment by thecombatwombat** [comment]
LGTM. Merging now.
